Initial implementation of JAX-RS 3.1 (still using RESTEasy for 3.0) and TCK FAT

andymc12 opened the pull request:
- This adds the 3.1 feature code and updates the autofeatures.
- It points to a snapshot build of the JAX-RS 3.1 APIs.
- In includes the snapshot of the TCK running as a FAT.
- All translation files copied over from the 3.0 implementation.
- The implementation is using the 3.0 (transformed) version of RESTEasy.

Part of epic issue #18739.

andymc12 committed:
Initial implementation of JAX-RS 3.1 (still using RESTEasy for 3.0)

- This adds the 3.1 feature code and updates the autofeatures.
- It points to a snapshot build of the JAX-RS 3.1 APIs.
- In includes the snapshot of the TCK running as a FAT.
- All translation files copied over from the 3.0 implementation.
- The implementation is using the 3.0 (transformed) version of RESTEasy.

andymc12 commented:
#build

ramkumar-k-9286 commented:
It seems like there are no English messages to review in this issue.

Adding the `ID Review not needed` label. If this is not correct please let me know.

Regards,
Ramkumar

andymc12 committed:
Include DHE repo for snapshot builds

andymc12 committed:
Code review comments

- cleaning up comments/unnecessary code in pom.xml
- removing unnecessary log4testng.properties file

skasund commented:
L2 message review not required

tevans78 reviewed (changes requested):
Please wait for the feature to at least be prioritzed before merging this.

tevans78 commented on dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restfulWS3.0-xmlBinding3.0.feature:
```diff
@@ -1,7 +1,7 @@
 symbolicName=io.openliberty.restfulWS3.0-xmlBinding3.0
 visibility=private
-IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.restfulWSClient-3.0))", \
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.restfulWSClient-3.0)(osgi.identity=io.openliberty.restfulWSClient-3.1)))", \
  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.xmlBinding-3.0))"
```
This PR should go in after https://github.com/OpenLiberty/open-liberty/pull/19534
That will make a `xmlBinding-4.0` feature should be available (even if it is still based on 3.0) and so this auto-feature should take that into account.

tevans78 commented on dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.jakarta.restfulWS-3.1.feature:
```diff
@@ -0,0 +1,19 @@
+IBM-Process-Types: client, \
+ server
+Subsystem-Name: Jakarta RESTful Web Services 3.1
+-features=com.ibm.websphere.appserver.eeCompatible-9.0, \
```
Similarly, #19534 will make `eeCompatible-10.0` available

tevans78 commented on dev/io.openliberty.jakarta.rest.3.1.internal_fat_tck/publish/servers/FATServer/server.xml:
```diff
@@ -0,0 +1,39 @@
+        <feature>localConnector-1.0</feature>
+
+        <!--  features under test -->
+        <feature>appSecurity-4.0</feature>
+        <feature>jsonb-2.0</feature>
+        <feature>restfulWS-3.1</feature>
+        <feature>ssl-1.0</feature>
+        <feature>xmlBinding-3.0</feature>
```
Use EE 10 features

andymc12 commented:
Last build failed due to missing an Apache HTTP Client 3.1 class - explicitly adding that into the pom.xml - new #build

andymc12 commented:
This currently cannot be merged because the `arquillian-support-jakarta-2.0` feature depends on `servlet-5.0`, and fails when testing with EE10 features. I opened https://github.com/OpenLiberty/liberty-arquillian/pull/114 which would resolve issue https://github.com/OpenLiberty/liberty-arquillian/issues/113 - once that issue is resolved and pulled in to Liberty, then development on this PR could continue.

jim-krueger commented:
Closing,  changes updated and merged under #20505
